Preferences Form and E2E Flow

tzada8 opened the pull request:
## Link to Ticket
https://trello.com/c/rQaFl6fC
https://trello.com/c/bss0dvan

## Brief Description of Changes
- Added modals for preferences and feature prioritization (I'm aware the code is redundant between the 2 modals; this is something I'll look to simplify over reading week)
- Connected E2E flow such that a user can start from search and end up on the comparison page
- Verified that all data is correctly passed through

NOTE: I'm still referencing the dummy endpoints, but I think this might be fine for PPC1. For PPC2 (when the app is hopefully fully completed), then we can reference actual endpoints

NOTE 2: I know Kejia shared the idea of moving the forms to the front of the flow, but since we don't have any designs for these yet I think it's fine to have them in the middle of the flow like we initially thought of. For PPC2 (assuming designs are completed), then I'll move the forms to the front of the flow.

## Checklist
- [X] I do not commit the .env file.
- [X] I added my changes to the appropriate frontend or backend folders.
- [X] All aspects of the ticket have been entirely addressed and completed.
- [ ] Thorough automated tests have been added.
- [X] Functionality has been validated through manual tests.
- [ ] I have received approval from a peer before merging.

kprazuch reviewed (approved):
Looks good! Just two things:

1. The (currently) final questions and answers are: Brand: [Canon, Nikon, Other], Number of Megapixels: [<15, 15-30, 30-45, >45], Lens Type: [Fixed, Standard, Other], Camera Type: [Point and Shoot, DSLR, Mirrorless, Other], Budget: [<750, 750-1500, >1500]
2. I know you said you'll do error-handling later, but just want to make sure that ranking the features will be unskippable.

tzada8 commented:
Thanks for providing the list of updated options here!
1. I updated the preferences form with the new options.
2. I made each of the fields in the feature prioritization section required. Something now needs to be provided to submit the form (still need to add error handling to ensure the user doesn't put the same feature for multiple rankings). Just wondering though, is there any point to the "Skip" now if this form is required for them to complete? 

I also changed some of the fields I had to be in unison with the backend (e.g. changed `numberOfMegapixels` to the `megapixels` field that you're expecting)
